pkgs/by-name/pd/pd/package.nix

```
@@ -0,0 +1,112 @@
+{ romID ? "ntsc-final"
+
+, lib
+
+, fetchFromGitHub
+, stdenv_32bit
+
+, pkgsi686Linux # SDL2 and zlib
+
+, pkg-config
+, python3
+}: let
+  roms = {
+    "ntsc-final" = {
+      binName = "pd";
+    };
+    "pal-final" = {
+      binName = "pd.pal";
+    };
+    "jpn-final" = {
+      binName = "pd.jpn";
+    };
+  };
+  binName = roms.${romID}.binName;
+in
+  assert lib.assertOneOf "romID" romID (builtins.attrNames roms);
+
+  stdenv_32bit.mkDerivation rec {
+  pname = "pd";
+  version = "0-unstable-2024-05-31";
+
+  src = fetchFromGitHub {
+    owner = "fgsfdsfgs";
+    repo = "perfect_dark";
+    rev = "b523b1e076158223ff5f07a843522a7898eabb91";
+    hash = "sha256-GGmmjwIXsxa1yQgXQEQe0C8RWFd5+Jx6BWrXosqIkOQ=";
+  };
+
+  buildInputs = with pkgsi686Linux; [
+    SDL2
+    zlib
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+    python3
+  ];
+
+  # the project uses git to retrieve version informations but our fetcher deletes the .git
+  # so we replace the commands with the correct data directly
+  postPatch = ''
+    sed -i 's/git rev-parse --short HEAD/echo ${builtins.substring 0 9 src.rev}/' Makefile.port
+    sed -i 's/git rev-parse --abbrev-ref HEAD/echo port/' Makefile.port
+  '';
+
+  enableParallelBuilding = true;
+
+  hardeningDisable = [ "format" ]; # otherwise fails to build
+  hardeningEnable = [ "pie" ];
+
+  makeFlags = [
+    "TARGET_PLATFORM=i686-linux" # the project is 32-bit only for now
+    "ROMID=${romID}"
+    "GCC_OPT_LVL=" # "-Og" is explicitly set as default in pd, we remove it to use stdenv's settings instead (defaults to "-O2")
+  ];
+
+  makefile = "Makefile.port";
+
+  preBuild = ''
+    patchShebangs .
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    mv build/${romID}-port/${binName}.exe $out/bin/${binName}
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```

```suggestion
  meta = {
```
https://github.com/NixOS/nixpkgs/issues/292468